### ISSUE TITLE
rz_rand64 does not limit result to `max` value

### ISSUE THREAD

**Issue by ret2libc** [rizinorg/rizin#3740]
```c
static ut64 rz_rand64(ut64 mod) {
#if HAVE_ARC4RANDOM_UNIFORM
	return (ut64)arc4random_uniform(mod) << 32 | (ut64)arc4random_uniform(mod);
#else
	return ((ut64)rand() << 32 | (ut64)rand()) % mod;
#endif
}
```

`mod` is supposed to be the `max` value for the random number.
So `rz_rand64(5)` should return numbers between 0 and 4 (included).

However, when HAVE_ARC4RANDOM_UNIFORM is defined, the number could be much larger than `max` because `arc4random_uniform` is called one time for the high-half of the 64bit number and another time for the low-half.

**Comment by p371jah** [comment]
Was it fixed by #3742?

**Comment by ret2libc** [comment]
Yeah, you are right! Closing!
(Happy to see you back @pelijah ! )
